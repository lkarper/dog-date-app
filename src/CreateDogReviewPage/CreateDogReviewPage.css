.CreateDogReviewPage__section.outer-section {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.CreateDogReviewPage__section.outer-section h1 {
    text-align: center;
}
