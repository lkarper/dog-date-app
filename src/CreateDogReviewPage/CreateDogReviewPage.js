import React, { useContext, useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import UserContext from '../contexts/UserContext';
import DogReviewForm from '../DogReviewForm/DogReviewForm';
import DogListView from '../DogListView/DogListView';
import DogProfilesService from '../services/dog-profiles-service';
import './CreateDogReviewPage.css';

const CreateDogReviewPage = (props) => {
    const { dog_id } = props.match.params;

    const context = useContext(UserContext);

    const [dog, setDog] = useState();

    useEffect(() => {
        window.scrollTo(0, 0);
        DogProfilesService.fetchDogProfileById(dog_id)
            .then(dog => {
                setDog(dog);
            })
            .catch(error => {
                console.log(error);
                setDog(null);
            })
    }, [props, dog_id]);

    if (dog === null) {
        return (
            <section 
                className='CreateDogReviewPage__section section'
                aria-live='polite'
            >
                <header>
                    <h1>Dog not found</h1>
                </header>
                <p>We couldn't find the dog that you're attempting to review. Check the address and your connection before trying again.</p>
                {context.packMembers.length 
                    ?             
                        <div>
                            <h2>Would you like to review one of your pack members?</h2>
                            <ul>
                                {context.packMembers.map(dog => 
                                    <div key={dog.id}>
                                        <DogListView dog={dog.profile} />
                                        <Link to={`/leave-review/${dog.profile.id}`}>Click here to review {dog.profile.name}</Link>
                                    </div>
                                )}
                            </ul>
                        </div>
                    :
                        '' 
                }
            </section>
        );
    }

    if (dog) {
        return (
            <section 
                className='CreateDogReviewPage__section outer-section'
                aria-live='polite'
            >
                <header>
                    <h1>Use this form to leave a review of {dog.name}</h1>
                </header>
                <DogReviewForm
                    dogName={dog.name}
                    dog_id={dog_id}
                />
            </section>
        );
    }

    if (dog_id === '') {
        return (
            <section 
                className='CreateDogReviewPage__section error'
                aria-live='polite'
            >
                <p>Looks like something went wrong.  Check your connection and the url and try again.</p>
            </section>
        );
    }

    return (
        <section 
            className='CreateDogReviewPage__section'
            aria-live='polite'
        >
            <p>Loading...</p>
        </section>
    );
}

CreateDogReviewPage.defaultProps = {
    match: {
        params: {
            dog_id: '',
        }
    }
};

CreateDogReviewPage.propTypes = {
    match: PropTypes.object,
};

export default CreateDogReviewPage;
