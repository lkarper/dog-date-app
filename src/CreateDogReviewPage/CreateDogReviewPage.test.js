import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CreateDogReviewPage from './CreateDogReviewPage';

describe('CreateDogReviewPage', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CreateDogReviewPage />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CreateDogReviewPage />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
