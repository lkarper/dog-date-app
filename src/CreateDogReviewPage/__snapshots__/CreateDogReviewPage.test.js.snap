// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDogReviewPage renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CreateDogReviewPage
      match={
        Object {
          "params": Object {
            "dog_id": "",
          },
        }
      }
    >
      <section
        aria-live="polite"
        className="CreateDogReviewPage__section error"
      >
        <p>
          Looks like something went wrong.  Check your connection and the url and try again.
        </p>
      </section>
    </CreateDogReviewPage>
  </Router>
</BrowserRouter>
`;
